### ISSUE TITLE
Error during installation, https://elpa.gnu.org/packages/spinner-1.7.3.el: Bad Request

### ISSUE THREAD

**Issue by bradyt** [emacs-lsp/lsp-java#142]
With a minimal config, I get an error on installation of `lsp-java`.

That is, if I run a brand new emacs session with the following config,

``` elisp
(setq package-archives
      '(("gnu" . "https://elpa.gnu.org/packages/")
        ("melpa" . "https://melpa.org/packages/")))
(package-initialize)

(unless package-archive-contents
  (package-refresh-contents))

(package-install 'lsp-java)
```

Then I get,

```
Importing package-keyring.gpg...
Importing package-keyring.gpg...done
Contacting host: elpa.gnu.org:443
Contacting host: elpa.gnu.org:443
Contacting host: melpa.org:443
Package refresh done
Setting `package-selected-packages' temporarily since "emacs -q" would overwrite customizations
Setting `package-selected-packages' temporarily since "emacs -q" would overwrite customizations
Generating autoloads for request.el...
Generating autoloads for request.el...done
Wrote /root/.emacs.d/elpa/request-20181129.1138/request-autoloads.el
Wrote /root/.emacs.d/elpa/request-20181129.1138/request-autoloads.el

[elided]

Generating autoloads for markdown-mode.el...
Generating autoloads for markdown-mode.el...done
Wrote /root/.emacs.d/elpa/markdown-mode-20190305.319/markdown-mode-autoloads.el
Wrote /root/.emacs.d/elpa/markdown-mode-20190305.319/markdown-mode-autoloads.el
Checking /root/.emacs.d/elpa/markdown-mode-20190305.319...
Compiling /root/.emacs.d/elpa/markdown-mode-20190305.319/markdown-mode-autoloads.el...
Compiling /root/.emacs.d/elpa/markdown-mode-20190305.319/markdown-mode-pkg.el...
Compiling /root/.emacs.d/elpa/markdown-mode-20190305.319/markdown-mode.el...
Done (Total of 1 file compiled, 2 skipped)
Contacting host: elpa.gnu.org:443
https://elpa.gnu.org/packages/spinner-1.7.3.el: Bad Request
```

It's also available as a reproduce in a Docker recipe, at the following commit https://github.com/bradyt/docker-emacs/tree/429a5fdcd97492bb189c930f29804f0a11c45bbb.

If I understand correctly, it's actually `lsp-mode` that is requiring spinner 1.7.3, but the issue does not reproduce if I only put `lsp-mode` in the packages to install.

I'm not sure this is really a bug in `lsp-java`, it's only that I see it here. Should I post this to `lsp-mode` or even file a bug report with Emacs?

**Comment by bradyt** [comment]
Oh if it's related more to `package.el`, then the emacs version may be relevant. Though it's implicit in the Docker image. But here it is for convenience.

```
GNU Emacs 26.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.22.30) of 2019-05-29
```

**Comment by yyoncho** [comment]
I am unable to repro this issue even when using the docker container. IMO you have hit elpa being down.

**Comment by yyoncho** [comment]
```
Checking /root/.emacs.d/elpa/spinner-1.7.3...
Compiling /root/.emacs.d/elpa/spinner-1.7.3/spinner-autoloads.el...
Compiling /root/.emacs.d/elpa/spinner-1.7.3/spinner-pkg.el...
Compiling /root/.emacs.d/elpa/spinner-1.7.3/spinner.el...
Done (Total of 1 file compiled, 2 skipped)
Contacting host: melpa.org:443
```

Here it is the log. I am closing the issue

**Comment by bradyt** [comment]
Confirmed, thank you!

**Comment by bradyt** [comment]
Hmm, I see the same error again. Did you try the exact repo recipe, with silex, etc?

**Comment by bradyt** [comment]
Hmm, this is happening for me again. Did you verify with my exact recipe? So it only happens the first time emacs is started. The second time, the lsp-java install succeeds.

**Comment by yyoncho** [comment]
@bradyt the issue is not related to the package - elpa goes offline from time to time.

**Comment by bradyt** [comment]
Did you verify with my exact recipe? When I run it to install just lsp-java, it fails on installing spinner (nineteen times out of twenty). But if I install just spinner, there's no issues (ever). That doesn't seem like simply, "elpa goes offline". I'm not saying it's got be an issue with lsp-java package, I'm just being particular about your declaring that it's simply elpa goes offline. Maybe package.el has a bug when dealing with transitive dependencies. I have no idea. Anyways, waiting to see my email show up on emacs bug mailing list.

**Comment by shybovycha** [comment]
Had same issue for quite some time, elpa does not seem to be down.

Replacing the URL for elpa from `https` to `http` solved the problem:

```
(setq package-archives '(("gnu" . "http://elpa.gnu.org/packages/")
                         ("melpa" . "http://melpa.org/packages/"))
```

**Comment by bradyt** [comment]
Removing https seems a bad solution.

**Comment by bradyt** [comment]
> Replacing the URL for elpa from `https` to `http` solved the problem:

Removing https is not a good solution. I would suggest trying #emacs on IRC, or one of the emacs mailing lists.

**Comment by wschenk** [comment]
This solution (https -> http) fixed this issue for me also, and I am also new to emacs also.

**Comment by PerMildner** [comment]
I got a similar symptom. The fix in syl20bnr/spacemacs#12535 solved it for me (TLS version issue). This also explains why using http instead of https is a workaround.

**Comment by yaitskov** [comment]
tls termination could be done on a local proxy which should be safe
